cosign/README.md

```diff
@@ -0,0 +1,125 @@
+
+## Scenarios
+
+`ratify-verifier-go` supports multiple verification scenarios based on different signing methods, artifact types, and trust sources[3]. These scenarios can be categorized into the following main types.
+
+| **Verification Scenario**         | **Purpose**                                                            | **Use Case**                                                                 |
+|-----------------------------------|------------------------------------------------------------------------|-----------------------------------------------------------------------------|
+| **Keyless Verification**          | Verifyies signatures from Rekor, trust material (signed certificate timestamp) from the CT log, and certificates that chain up to Fulcio. | CI/CD pipelines or automated workflows where no private key management is needed. |
+| **Key-Based Verification**        | Verifies signature using a known **public key**.                        | Environments where signatures are verified with a known public key. |
+| **Timestamp Verification**        | Verifies the **timestamp** of the signature to prevent time-based attacks. Not standalone verification. | Long-term signature validity checks, ensuring signatures are valid at a specific point in time. |
+| **Rekor Transparency Log (TLog)** | Verifies inclusion of the signature in the **Rekor Transparency Log** for audibility. Not standalone verification. | Auditing and compliance to ensure signatures are publicly recorded in an immutable log. |
+
+### Cosign Verifier Input
+
+1. The Artifact to Verify
+    - The artifact in OCI-compliant registries.
+
+2. Verification Materials
+    - The signature(s) of the artifact[4]: The cryptographic signature(s) created when the artifact was signed.
+    - Verification options: i.e. whether to expect signed certificate timestamps (SCTs), TLog entries, or signed timestamps and expected identity and digest to verify.
+
+3. Verification Key or Certificate
+    - Keyless Verification: Retrieves an X.509[5] certificate on the signature and verify against Fulcio root trust.
+    - Key-Based Verification: User provides a public key, certificate or certificate chain.
+
+### Cosign Verifier Output
+
+The output format is defined by `ratify-go`, illustrates if the signature is valid and all criteria are met. It includes detailed information such as:
+
+- verification_success: Indicates if the verification was successful.
+- verifier reference:  The verifier that generated the result.
+- artifact_digest: The cryptographic hash of the artifact.
+- signer_identity: The identity associated with the signing certificate (if keyless signing was used).
+- signing_cert: Metadata about the Fulcio-issued certificate (if applicable).
+- rekor_entry: Confirms whether the signature exists in the transparency log (for keyless verification).
+- timestamp_verified: Ensures that the signature has a valid timestamp (if timestamping was used).
```

Comment:
Are they used by this ratify verifier?

Reply:
Reference: https://pkg.go.dev/github.com/ratify-project/ratify-go@main#VerificationResult

Reply:
I am planning to have all these details showing in the `detail` section of the result, we can discuss and make an update later.